src/App.tsx

```diff
@@ -34,62 +37,122 @@ export function getReorderedGoods(
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  isChanged: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+    isChanged: false,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortType: SortType.ALPABET,
+      isChanged: true,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      isChanged: true,
+    });
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  resetOptions = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      isChanged: false,
+    });
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info',
+              { 'is-light': this.state.sortType !== SortType.ALPABET },
+            )}
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-success',
+              { 'is-light': this.state.sortType !== SortType.LENGTH },
+            )}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-warning',
+              { 'is-light': this.state.isReversed !== true },
+            )}
+            onClick={this.reverseList}
+          >
+            Reverse
+          </button>
+
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={this.resetOptions}
+            style={(this.state.isChanged || this.state.isReversed)
+              ? { visibility: 'visible' }
+              : { visibility: 'hidden' }}
```

Comment:
move condition to separate var, use class names and 
https://bulma.io/documentation/helpers/visibility-helpers/#other-visibility-helpers